examples/real_function_space.py

```diff
@@ -115,26 +115,60 @@ def u_exact(x):
 
 # We can now assemble the matrix and vector
 
-A = dolfinx.fem.petsc.assemble_matrix_block(a)
+try:
+    A = dolfinx.fem.petsc.assemble_matrix_block(a)
+except AttributeError:
+    A = dolfinx.fem.petsc.assemble_matrix(a, kind="mpi")
 A.assemble()
-b = dolfinx.fem.petsc.assemble_vector_block(L, a, bcs=[])
+
+
+# On the main branch of DOLFINx, the `assemble_vector` function for blocked spaces has been rewritten to reflect how
+# it works for standard assembly and `nest` assembly. This means that lifting is applied manually.
+# In this case, with no Dirichlet BC, we could skip those steps.
+# However, for clarity we include them here.
+
+bcs = []
+main_assembly = False
+try:
+    b = dolfinx.fem.petsc.assemble_vector_block(L, a, bcs=bcs)
+except AttributeError:
+    main_assembly = True
+    b = dolfinx.fem.petsc.assemble_vector(L, kind="mpi")
+    bcs1 = dolfinx.fem.bcs_by_block(dolfinx.fem.extract_function_spaces(a, 1), bcs)  # type: ignore
```

Comment:
This is something new that I don't understand. Would it be possible to add some more documentation here or refer to the documentation in dolfinx?